### DISCUSSION TITLE
After running a cell a white dot appears on the file tab.

### DISCUSSION THREAD

**Discussion by smithy49** [microsoft/vscode-jupyter#10762] [Questions and Answers] [answered]
Hi,
I'm running VSCode 1.69 on Windows 10.
When running certain cells a white dot appears in the file tab.
After saving the file the dot disappears and re-appears after running the cell again.
This happens when the code below is executed.

`''.join(["hello", "world"])`

Thanks for any help.

**Comment by amunger** [comment, marked as answer]
the execution count is stored in the text of the ipynb, so every execution will change the underlying file. If I know which white dot you are referring to, that just means the file has unsaved changes.

**Reply by smithy49** [reply]
Thanks for replying.
I've attached a snap shot of the white dot.

![jupyter](https://user-images.githubusercontent.com/6471037/178357357-98d907ef-68ec-422b-a7fb-14e9abde3d5f.JPG)

**Reply by amunger** [reply]
I haven't seen that before, do you have another extension that could be adding it? From the behavior in the original description, it sounds like a dirty indicator, but I don't know why it would show there.
Is it only for specific files (.ipynb) and only for executing that specific code?

**Reply by oscar-defelice** [reply]
yes, that one is the "unsaved changes" sign.
A ipynb file is ultimately a json file. There is an execution counter field for each cell. So if you run a cell you are changing the file.

**Reply by smithy49** [reply]
I've been doing some experimentation and this is what I've discovered.
I tried VSCode on Ubuntu 20.04 and executed the same cell and the dirty indicator show up momentarily which implies that the file was auto saved.
I then installed VSCode from scratch on Windows 10 in a virtual machine and running the same cell produced the dirty indicator until I manually saved the file. Should I raise a bug?

**Reply by amunger** [reply]
auto-save is off by default, so that sounds like it's working as intended as long as your Ubuntu install has auto-save enabled. `Files: auto save` in settings or `files.autoSave` in the settings.json

**Reply by smithy49** [reply]
@amunger Thanks for your tip. The dirty indicator now gone after setting auto-save.
